src/peppr/rotamer_rama.py

```diff
@@ -0,0 +1,1259 @@
+    )
+    classification: str = Field(
+        description="Rotamer classification: FAVORED, ALLOWED, OUTLIER, UNKNOWN",
+        examples=["FAVORED", "ALLOWED", "OUTLIER", "UNKNOWN"],
+    )
+    error: str | None = Field(
+        description="Error message if any error occurred during rotamer scoring",
+        default=None,
+    )
+
+    @classmethod
+    def from_residue(
+        cls, atom_array: struc.AtomArray, res_id: int, chain_id: str, model_no: int
+    ) -> "ResidueRotamerScore":
+        """
+        Create ResidueRotamerScore from a residue object.
+
+        Parameters
+        ----------
+        atom_array : struc.AtomArray
+            The AtomArray containing the structure.
+        res_id : int
+            The residue ID to check.
+        chain_id : str
+            The chain ID to check.
+        model_no : int
+            The model number to check (usually 0 for single model structures).
+
+        Returns
+        -------
+        ResidueRotamerScore
+            An instance of ResidueRotamerScore containing the rotamer score for the residue.
+        """
+        result = check_rotamer(atom_array, res_id, chain_id, model_no=model_no)
+        return cls(**result)
+
+
+class RotamerScore(BaseModel):
+    """
+    Rotamer score for a given protein structure.
+    This class represents the rotamer scores for all residues in a protein structure,
+    including their names, IDs, observed chi angles, rotamer score percentages, and classifications.
+    """
+
+    model_config = ConfigDict(from_attributes=True)
+    rotamer_scores: list[ResidueRotamerScore] | list[list[ResidueRotamerScore]] = Field(
+        description="List of rotamer scores for each residue in the structure."
+    )
+
+    @classmethod
+    def from_atom_array(
+        cls, atom_array: struc.AtomArray, model_no: int = 0
+    ) -> "RotamerScore":
+        """
+        Create ResidueRotamerScore from a protein structure.
+
+        Parameters
+        ----------
+        atom_array : struc.AtomArray
+            The AtomArray containing the structure.
+        model_no : int, optional
+            The model number to use for classification (default is 0).
+
+        Returns
+        -------
+        RotamerScore
+            An instance of RotamerScore containing the rotamer scores for the structure.
+        """
+        if not isinstance(atom_array, struc.AtomArray):
+            LOG.warning(
+                f"RotamerScore: Cannot be computed for model {model_no}; atom_array must be a biotite structure AtomArray"
+            )
+            return cls(rotamer_scores=[])
+
+        if all(atom_array.hetero):
+            LOG.warning(
+                f"RotamerScore: Cannot be computed for model {model_no}; atom_array must contain some protein residues"
+            )
+            return cls(rotamer_scores=[])
+        atom_array = atom_array[
+            struc.filter_amino_acids(atom_array) & ~atom_array.hetero
+        ]
```

Comment:
`filter_amino_acids()` will also filter non-canonical amino acids, but for them the chi angles above are not defined, right? If this is true, this would be the alternative.
```suggestion
        atom_array = atom_array[
            struc.filter_canonical_amino_acids(atom_array) & ~atom_array.hetero
        ]
```